composer.json

```diff
@@ -17,7 +17,7 @@
         "php": "^5.6 || ^7.0",
         "ext-json": "*",
         "ext-tokenizer": "*",
-        "composer/semver": "^1.4",
+        "composer/semver": "^1.4 || ^2.0",
```

Comment:
Let's have `^3.0` allowed too.

Reply:
would be great to test if we work with v1, v2 and v3, if we claim to support all

Reply:
The need to have specific tests for every supported version is the proof that widening constraints increase maintainance work. Let's do the opposite and restrict dependencies constraints as much as possible. This is related to #4810.